### PR TITLE
Split out template and content selection

### PR DESCRIPTION
**Pull request by alex-deaconu**
## Description
Split out template and content selection

Create content selection component
Create simple service to overwrite products factory

Add component and dependencies to CH build

[stage-2]

## Motivation and Context
Allow RVA to open the Store Content modal independently of Store; Deprecate Store UI

## How Has This Been Tested?
Tested changes locally. Updated unit tests. Updated and tested RVA.

## Release Plan:
- As the Submitter, upon requesting review of this pull request, I confirm that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed. 
- As the Reviewer, upon approving the changes in this PR, I confirm I have reviewed and I agree that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed

#### Release Checklist Items Skipped?
No

### CODE REVIEW

**Comment by alex-deaconu** [comment]
@ezequielc please review. Thanks!

File: test/unit/editor/controllers/ctr-store-products-modal.tests.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 'use strict';
 
-describe('controller: Store Products Modal', function() {
+describe('controller: Store Templates Modal', function() {
```
Comment (ezequielc) [inline test/unit/editor/controllers/ctr-store-products-modal.tests.js:3]: @alex-deaconu I think this file should be rename to `ctr-store-templates-modal.tests.js`. Thanks

File: test/unit/components/store-products/ctr-store-products-modal.tests.js [hunk 2]
```diff
@@ -0,0 +1,169 @@
+'use strict';
+
+describe('controller: Store Content Modal', function() {
```
Comment (ezequielc) [inline test/unit/components/store-products/ctr-store-products-modal.tests.js:3]: @alex-deaconu The unit test file name does not match here as well. Thanks

**Review by ezequielc — APPROVED** [review]
@alex-deaconu Looks great. Small comment inline. Thanks

**Comment by ezequielc** [comment]
@alex-deaconu Thanks for the updates 👍

**Pull request merged**